### PR TITLE
To support DTP2023.1 for geting rule doc url without auth

### PR DESCRIPTION
**Pull request by rwang-parasoft**

### CODE REVIEW

File: PublishParasoftTestResults/publishtestresults.ts [hunk 1]
```diff
@@ -539,18 +517,31 @@ function appendRuleDocUrls(sarifReport: string) {
...
-        tl.warning("Unable to connect to DTP to retrieve documentation for rules using the provided settings: Error code " + (error.response ? error.response.status : undefined));
-    });
+function verifyDtpRuleDocsService() {
+    return tryToGetNotExistingRuleDocUrl()
+            .then(() => {
+                // Should never reach this block unless there is a `notExistingRule` rule id
+                // and `notExistingAnalyzerId` analyzer id pair in the future.
+                isDtpRuleDocsServiceAvailable = true;
+            }).catch((error) => {
+                const status =  error && error.response && error.response.data ? error.response.data.status : undefined;;
```
Comment (yshi-parasoft) [inline PublishParasoftTestResults/publishtestresults.ts:527]: Remove the double semicolon in the end.

File: PublishParasoftTestResults/publishtestresults.ts [hunk 2]
```diff
@@ -539,18 +517,31 @@ function appendRuleDocUrls(sarifReport: string) {
...
+                if (status == 404) {
+                    isDtpRuleDocsServiceAvailable = true;
+                } else if (status == 401) {
+                    // Need auth to get doc url for DTP version below 2023.1.
+                    isDtpRuleDocsServiceAvailable = false;
+                    tl.warning("Unable to retrieve documentation for rules from current version of DTP. DTP 2023.1 or newer version is needed.");
```
Comment (yshi-parasoft) [inline PublishParasoftTestResults/publishtestresults.ts:533]: Change to:
Unable to retrieve the documentation for the rules from the current version of DTP. DTP  2023.1 or a newer version is required.

File: PublishParasoftTestResults/publishtestresults.ts [hunk 3]
```diff
@@ -539,18 +517,31 @@ function appendRuleDocUrls(sarifReport: string) {
...
+                } else {
+                    isDtpRuleDocsServiceAvailable = false;
+                    tl.warning("Unable to connect to DTP to retrieve documentation for rules using the provided settings: Error code " +
```
Comment (yshi-parasoft) [inline PublishParasoftTestResults/publishtestresults.ts:536]: Change to:
Unable to connect to DTP and retrieve the documentation for rules using the provided settings (error code: xxx). Please make sure the values for "dtp.*" in xxx are correct.

File: PublishParasoftTestResults/publishtestresults.ts [hunk 4]
```diff
@@ -539,18 +517,31 @@ function appendRuleDocUrls(sarifReport: string) {
...
+                                status + ". Check the dtp.* values in " + localSettingsPath);
+                }
+            });
+}
+
+/**
+ * 404 is expected when response is returned as normal.
+ */
+function tryToGetNotExistingRuleDocUrl() {
```
Comment (yshi-parasoft) [inline PublishParasoftTestResults/publishtestresults.ts:545]: If this method is only used once, I think it's better to make the call directly in "verifyDtpRuleDocsService" and just add reasonable comment.

**Comment by rwang-parasoft** [comment]
I have created another pull request since the code structure has been refactored.

**Pull request closed without merging**